bin/draw_l1analysis.cxx

```diff
@@ -0,0 +1,421 @@
+    meteffHists_new_cond[metType]->SetMarkerColor(kRed);
+
+    meteffHists_def[metType]->SetMarkerSize(0.8);
+    meteffHists_new_cond[metType]->SetMarkerSize(0.8);
+    
+    meteffHists_def[metType]->Draw("AP");
+    meteffHists_new_cond[metType]->Draw("P");
+    meteffHists_def[metType]->GetXaxis()->SetTitle("Reco MET (GeV)");
+    meteffHists_def[metType]->GetYaxis()->SetTitle("Efficiency");
+    
+    tcanvases.back()->Print(Form("plots/%smetEffs_emu.pdf", metType.c_str()));
+    
+
+  }
+
+   // Other res plots
+  
+  TF1 *fgaus = new TF1("g1","gaus");//,-2.,2.);
+  fgaus->SetRange(-1.,1.);
+  
+  // // Jet/MET resolution
+  // std::vector<std::string> resTypes = {"hresJet","hResMET"};
+  std::vector<TCanvas*> mycanvases;
+  TH2F *resJet_def = dynamic_cast<TH2F*>(files.at(0)->Get("hresJet"));resJet_def->RebinX(4);
+  TH2F *resJet_new_cond = dynamic_cast<TH2F*>(files.at(1)->Get("hresJet"));resJet_new_cond->RebinX(4);
+
+  resJet_def->FitSlicesY(fgaus);
+  TH1D *resJet_def_1 = (TH1D*)gDirectory->Get("hresJet_1");
+  TH1D *resJet_def_2 = (TH1D*)gDirectory->Get("hresJet_2");
+  gROOT->cd();
+  
+  resJet_new_cond->FitSlicesY(fgaus);
+  TH1D *resJet_new_cond_1 = (TH1D*)gDirectory->Get("hresJet_1");
+  TH1D *resJet_new_cond_2 = (TH1D*)gDirectory->Get("hresJet_2");
+  gROOT->cd();
+  
+  mycanvases.push_back(new TCanvas);
+  mycanvases.back()->SetWindowSize(mycanvases.back()->GetWw(), 1.3*mycanvases.back()->GetWh());
+
+  gPad->SetGridx(); gPad->SetGridy();
+  
+  resJet_def_1->Draw("");
+  resJet_def_1->GetXaxis()->SetTitle("Reco Jet E_{T} (GeV)");
+  resJet_def_1->GetYaxis()->SetRangeUser(-1.5,1.5);
+  resJet_def_1->SetMarkerSize(0.5);
+  resJet_def_1->SetMarkerStyle(24);
+  
+  // mycanvases.push_back(new TCanvas);
+  // mycanvases.back()->SetWindowSize(mycanvases.back()->GetWw(), 1.3*mycanvases.back()->GetWh());
+ 
+  resJet_new_cond_1->Draw("same");
+  resJet_new_cond_1->SetLineColor(2);
+  resJet_new_cond_1->SetMarkerColor(2);
+  resJet_new_cond_1->SetMarkerSize(0.5);
+  resJet_new_cond_1->SetMarkerStyle(20);
+  mycanvases.back()->Print(Form("plots/%s_emu.pdf", "resJet_1"));
+
+  mycanvases.push_back(new TCanvas);
+  mycanvases.back()->SetWindowSize(mycanvases.back()->GetWw(), 1.3*mycanvases.back()->GetWh());
+  // resJet_def->FitSlicesY(fgaus);
+
+   gPad->SetGridx(); gPad->SetGridy();
+  
+  resJet_def_2->Draw("");
+  resJet_def_2->GetXaxis()->SetTitle("Reco Jet E_{T} (GeV)");
+  resJet_def_2->GetYaxis()->SetRangeUser(0.,0.9);
+  resJet_def_2->SetMarkerSize(0.5);
+  resJet_def_2->SetMarkerStyle(24);
+
+  // resJet_new_cond->FitSlicesY(fgaus);
+ 
+  resJet_new_cond_2->Draw("same");
+  resJet_new_cond_2->SetLineColor(2);
+  resJet_new_cond_2->SetMarkerColor(2);
+  resJet_new_cond_2->SetMarkerSize(0.5);
+  resJet_new_cond_2->SetMarkerStyle(20);
+  mycanvases.back()->Print(Form("plots/%s_emu.pdf", "resJet_2"));
+
+  // // MET resolution
+  // TF1 *fgaus2 = new TF1("g1","gaus");//,-2.,2.);
```

Comment:
Should this section of code be uncommented? If the code is needed only in certain cases, please allow configuration--for example, from the command line--rather than require recompilation.

Reply:
hi @christopheralanwest 
in principle it should work, in all cases . I plan to uncomment it, but I need to figure out first why the resolution summary plots are not displayed correctly . I will check that asap yes.